src/session/chat/history.rs

```diff
@@ -131,18 +131,12 @@ impl History {
         do_async(
             glib::PRIORITY_DEFAULT_IDLE,
             async move {
-                functions::GetChatHistory::new()
-                    .chat_id(chat_id)
-                    .from_message_id(oldest_message_id)
-                    .limit(20)
-                    .send(client_id)
+                functions::get_chat_history(chat_id, oldest_message_id, 0, 20, false, client_id)
                     .await
             },
             clone!(@weak self as obj => move |result| async move {
                 if let Ok(enums::Messages::Messages(result)) = result {
-                    if let Some(messages) = result.messages {
-                        obj.prepend(messages);
-                    }
+                    obj.prepend(result.messages);
```

Comment:
This should actually be an Option, maybe? The pattern for the nullable parameter is a bit different, I asked to tdlib devs, let's wait for the response to decide.

Reply:
The response is that in this case the items of the vector may be null, not thr vector itself. This should be handled by tdgrand.